tests/unit/dataactvalidator/test_c23_award_financial_1.py

```diff
@@ -0,0 +1,51 @@
+from random import randint, choice
+from string import ascii_uppercase, ascii_lowercase, digits
+from tests.unit.dataactcore.factories.staging import AwardFinancialFactory, AwardProcurementFactory
+from tests.unit.dataactvalidator.utils import number_of_errors, query_columns
+
+_FILE = 'c23_award_financial_1'
+
+def test_column_headers(database):
+    expected_subset = {"row_number", "transaction_obligated_amou_sum", "federal_action_obligation_sum"}
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset <= actual
+
+def test_success(database):
+    """ Test that a four digit object class with no flag is a success, and a three digit object class with a flag is a success"""
+    # Create a 12 character random piid
+    piid = ''.join(choice(ascii_uppercase + ascii_lowercase + digits) for i in range(12))
+    piid_two = ''.join(choice(ascii_uppercase + ascii_lowercase + digits) for i in range(12))
+    first_piid_row_one = AwardFinancialFactory(object_class = randint(1100,1999),
+        by_direct_reimbursable_fun = "d", transaction_obligated_amou = 1234, piid = piid)
```

Comment:
Removed these.  I had them there with separate ranges because a repeat of object class, program activity, that flag, and award ID is an error, but since that error is defined in a different rule it shouldn't affect this test, so no reason to specify them here.